src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -180,7 +180,7 @@ public static function get_core_data() {
 		 *
 		 * @param WP_Theme_JSON_Data Class to access and update the underlying data.
 		 */
-		$theme_json   = apply_filters( 'theme_json_default', new WP_Theme_JSON_Data( $config, 'default' ) );
+		$theme_json   = apply_filters( 'theme_json_default_data', new WP_Theme_JSON_Data( $config, 'default' ) );
```

Comment:
Another idea might be to follow a `theme_json_data_{$context}` syntax:

```suggestion
		$theme_json   = apply_filters( 'theme_json_data_default', new WP_Theme_JSON_Data( $config, 'default' ) );
```